### ISSUE TITLE
Connection failed on iOS

### ISSUE THREAD

**Issue by amserhii11** [taskrabbit/react-native-zendesk-chat#105]
On android it works fine. But when I send message from iOS  I can't get it on Zendesk and got this:
1) Last message can't be sent
2) After relaunch the bot we got "connection failed"

And can't retry connection. 
Any ideas? 
![image](https://user-images.githubusercontent.com/27774005/108356477-dbe35600-71f4-11eb-8ac0-1e5314fd74ac.png)

![image](https://user-images.githubusercontent.com/27774005/108356412-c706c280-71f4-11eb-97e6-ce033cee70c6.png)

**Comment by amserhii11** [comment]
@fbartho  Hi! Do you know what's the problem with it?  
RN 0.63.1
Lib 0.4.1-beta.7

**Comment by fbartho** [comment]
I just verified that it’s working fine for our company as of 10:30am California.

Is it possible this was a transient outage on Zendesk’s side, maybe due to the weather emergency in Texas? Or have you verified that live chat works on your website’s zendesk integration?

**Comment by amserhii11** [comment]
@fbartho  Hi. Sorry for the answer long.  I have verified that live chat works on android properly.  This issue only on iOS. I just can't get my messages on the Zendesk console. May it be my mistakes with zen desk configuration. Could I be missing something in the settings of chat ?

**Comment by amserhii11** [comment]
@fbartho  I just use only Zendesk Account Key without AppID. May it be the reason?

**Comment by fpena** [comment]
I have this issue also. Using both account key and app id.

**Comment by amserhii11** [comment]
@fpena  did you fix it?

**Comment by fpena** [comment]
> @fpena did you fix it?

For me it was just an error in the account key.

**Comment by amserhii11** [comment]
@fpena  What is the problem with it? Did not you configure the app in zendesk? Or just a mistake with a key? Because I still have it not fixed.

**Comment by rahulmr-rently** [comment]
Without appID will this work ? I am still getting this issue and keys are correct here

**Comment by amserhii11** [comment]
@rahulmr-rently  For me, it not works with appID as well.

**Comment by jamie78** [comment]
Facing same problem. This is probably zendesk issue. I have 2 accounts. One is working just fine and the other is not working on iOS. Just not connect to Zendesk.

**Comment by amserhii11** [comment]
@jamie78 Could you compare these 2 accounts? May it be problems with configuration Zendesk account? 
Do you use appID ? Because I tried both and it still not works for me.

**Comment by amserhii11** [comment]
@fpena Is it require to have PEM file in app configuration? I can't find info about it. May it be a problem with it?

**Comment by fpena** [comment]
> @fpena Is it require to have PEM file in app configuration? I can't find info about it. May it be a problem with it?

Not really. 
In order for me to work, was that I needed both account id and app id. All of this information can be seen with the Chat app in Zendesk.

**Comment by fpena** [comment]
> Without appID will this work ? I am still getting this issue and keys are correct here

It didn't work me without `appId`.

**Comment by amserhii11** [comment]
@fpena  Thank you. I tried all solutions but it still not works.
So sad.

**Comment by fpena** [comment]
> @fpena Thank you. I tried all solutions but it still not works.
> So sad.

Did you create the app in the Chats section? 
Also, I'm assigning some data after initializing the instance. Like name, phone and email I think.

**Comment by amserhii11** [comment]
@fpena  If you mean this one - I created it. Will try to assign data in 5 min.
![image](https://user-images.githubusercontent.com/27774005/110958767-b78f1b00-8355-11eb-8c83-c6723fbf71c2.png)

**Comment by jamie78** [comment]
I don't think you need appId. This is just for push notifications. I think Zendesk is having some nice bugs there. Can you try to create new account on zendesk and try it?

**Comment by amserhii11** [comment]
@jamie78  I tried 2 accounts. Both results for iOS.

**Comment by amserhii11** [comment]
I tried to check it on Swift sample project and it works properly. So, I think problems in the library.

**Comment by amserhii11** [comment]
Ok, I just got some problems with init params. I found it by
`- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
{
  #if DEBUG
    ZDKChatLogger.isEnabled = YES;
  #endif
 
    ... 
  
  return YES;
}`

Fixed.  Thanks all for helping.

**Comment by jamie78** [comment]
@amserhii11 what exactly was wrong? :)

**Comment by Giosanu** [comment]
@amserhii11 hey, can you give more details on your fix? I seem to be having the same problem (even thou two days ago it worked fine.. and I did nothing). Thanks!

**Comment by fbartho** [comment]
If you didn't change anything @Giosanu -- I would reach out to your Zendesk customer support representative. Maybe it's a known issue on their side!

**Comment by amserhii11** [comment]
For me it was a problem was in "Department" options. I am not sure why it happened, but I just change the department and now it works fine.

**Comment by ManishaMShetty** [comment]
Is this issue fixed ? I too facing the same issue with iOS device

**Comment by jamie78** [comment]
Hey, I was able to fix this issue by not using those lines:
```
    tags: ['Not-working'],
    department: 'Not working',
```

Pls check last message to see which issue I was exactly facing, here: https://support.zendesk.com/hc/en-us/community/posts/1500000092482-Connection-error-chat-SDK-iOS-Flutter

**Comment by shawnFallon** [comment]
@jamie78 Thanks, I was racking my brain all last night!  Removed the department and it all started working

**Comment by Nevosis** [comment]
@shawnFallon Thanks a lot, I don't know how or why, but it works
